### ISSUE TITLE
Crash while loading banner ads.

### ISSUE THREAD

**Issue by usman-whizpool** [googleads/googleads-mobile-flutter#895]
Flutter package google_mobile_ads: ^3.0.0

Loading ads create crash some times. unable to catch it. 

as per firebase logs

## Steps to Reproduce

here is sample code we prepared:

```dart
class AdScreenA extends StatelessWidget {
  const AdScreenA({super.key});

  @override
  Widget build(BuildContext context) {
    return SafeArea(
        child: Scaffold(
      appBar: AppBar(
        title: const Text(
          "Screen A",
          style: TextStyle(color: Colors.white),
        ),
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
        children: [
          const BannerAdMobContainer(),
          Center(
            child: GestureDetector(
              onTap: () {
                Get.to(() => const AdScreenB());
              },
              child: Container(
                width: 200,
                height: 40,
                color: Colors.red,
                child: const Center(
                    child: Text(
                  "Screen B",
                  style: TextStyle(color: Colors.white),
                )),
              ),
            ),
          ),
          const MidRectAdContainer()
        ],
      ),
      bottomNavigationBar: const BannerAdMobContainer(),
    ));
  }
}

class AdScreenB extends StatelessWidget {
  const AdScreenB({super.key});

  @override
  Widget build(BuildContext context) {
    return SafeArea(
        child: Scaffold(
      backgroundColor: Colors.blue,
      appBar: AppBar(
        title: Text(
          "Screen B",
          style: TextStyle(color: Colors.white),
        ),
      ),
      body: Column(
        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
        children: [
          const BannerAdMobContainer(),
          Center(
            child: GestureDetector(
              onTap: () {
                Get.back();
              },
              child: Container(
                width: 200,
                height: 40,
                color: Colors.red,
                child: const Center(
                    child: Text(
                  "Go Back",
                  style: TextStyle(color: Colors.white),
                )),
              ),
            ),
          ),
          const MidRectAdContainer()
        ],
      ),
      bottomNavigationBar: const BannerAdMobContainer(),
    ));
  }
}

class BannerAdMobContainer extends StatefulWidget {
  const BannerAdMobContainer({
    Key? key,
  }) : super(key: key);

  @override
  State<BannerAdMobContainer> createState() => _BannerAdMobContainerState();
}

class _BannerAdMobContainerState extends State<BannerAdMobContainer> {
  BannerAd? bannerAd;

  bool isLoaded = false;
  //-------------------------------------------
  @override
  void initState() {
    WidgetsBinding.instance.addPostFrameCallback((_) {
      loadAd();
    });
    super.initState();
  }

  //-------------------------------------------
  @override
  void dispose() {
    if (bannerAd != null) {
      bannerAd!.dispose();
    }
    super.dispose();
  }

  //-------------------------------------------
  @override
  Widget build(BuildContext context) {
    return !isLoaded || bannerAd == null
        ? const SizedBox(
            height: 50,
          )
        : SizedBox(
            width: double.infinity, //this is for giving infinite width
            height: bannerAd!.size.height.toDouble(),
            child: AdWidget(
              ad: bannerAd!,
            ),
          );
  }

  //-------------------------------------------
  void loadAd() async {
    if (kDebugMode) {
      print("loadAd: load ad one 1");
    }
    if (isLoaded || bannerAd != null) {
      return;
    }
    if (kDebugMode) {
      print("loadAd: load ad 2 bannerAdSize");
    }

    final AnchoredAdaptiveBannerAdSize? bannerAdSize =
        await AdSize.getCurrentOrientationAnchoredAdaptiveBannerAdSize(
      Get.width.truncate(),
    );
    if (kDebugMode) {
      print("loadAd: load ad 3 bannerAdSize");
    }

    bannerAd = BannerAd(
      adUnitId: kDebugMode ? AdsServices.debguBannerId : AdsServices.bannerid,
      request: const AdRequest(),
      size: bannerAdSize!,
      listener: BannerAdListener(
        onAdLoaded: (ad) {
          //this isLoaded booleans are important to display ads dont ever remove it
          setState(() {
            isLoaded = true;
          });
        },
        onAdFailedToLoad: (ad, err) {
          //this isLoaded booleans are important to display ads dont ever remove it
          setState(() {
            isLoaded = false;
          });
          ad.dispose();
        },
      ),
    )..load();
    if (kDebugMode) {
      print("loadAd: load ad one bannerAdSize loaded");
    }
  }
}

class MidRectAdContainer extends StatefulWidget {
  const MidRectAdContainer({
    Key? key,
  }) : super(key: key);

  @override
  State<MidRectAdContainer> createState() => _MidRectAdContainerState();
}

class _MidRectAdContainerState extends State<MidRectAdContainer> {
  BannerAd? bannerAd;
  bool isLoaded = false;

  @override
  void initState() {
    loadAd();
    super.initState();
  }

  @override
  void dispose() {
    if (bannerAd != null) {
      bannerAd!.dispose();
    }
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return !isLoaded
        ? SizedBox(
            width: CommonMethods.isTablet ? 450 : 320,
            height: CommonMethods.isTablet ? 300 : 200,
            //color: Colors.yellow,
          )
        : Center(
            child: SizedBox(
              width: bannerAd!.size.width.toDouble(),
              height: bannerAd!.size.height.toDouble(),
              child: AdWidget(
                ad: bannerAd!,
              ),
            ),
          );
  }

  void loadAd() async {
    // await Future.delayed(const Duration(seconds: 3));
    // if (kReleaseMode) {
    //   CommonMethods.showToast("temp disabled mid ad container");
    //   return;
    // }
    if (isLoaded) {
      return;
    }
    bannerAd = BannerAd(
      adUnitId: kDebugMode ? AdsServices.debguBannerId : AdsServices.bannerid,
      request: const AdRequest(),
      size: CommonMethods.isTablet
          ? const AdSize(
              width: 450,
              height: 300,
            )
          : const AdSize(
              width: 300,
              height: 200,
            ),
      listener: BannerAdListener(
        // Called when an ad is successfully received.
        onAdLoaded: (ad) {
          setState(() {
            isLoaded = true;
          });
        },
        // Called when an ad request failed.
        onAdFailedToLoad: (ad, err) {
          // if (kDebugMode) {
          //   print("onAdFailedToLoadMid===>${err.toString()}");
          // }
          setState(() {
            isLoaded = false;
          });
          ad.dispose();
        },
      ),
    )..load();
  }
}
```

**Comment by huycozy** [comment]
Hi @usman-whizpool 
Please provide the stack trace of the crash so that we may verify the issue properly. And which platform are you facing the crash?

Also, your sample code above is complicated and is production code that is not easy for us to investigate the issue, please provide a complete & minimal sample code without using 3rd package or complex code sample (You may try using [plugin sample code](https://github.com/googleads/googleads-mobile-flutter/tree/main/packages/google_mobile_ads/example) to check if you can reproduce the issue as well)

Thanks!

**Comment by usman-whizpool** [comment]
Here is stack trace. facing on iOS

**Comment by huycozy** [comment]
Thanks. Taking a look at that stack trace, Flutter frames are not symbolized yet. Are you only seeing the crash on the production? Can you try running the project in [release mode](https://docs.flutter.dev/testing/build-modes#release) to see if you can reproduce it?

**Comment by usman-whizpool** [comment]
replicated in example project also. added one more screen and navigated to it from existing screen with button click. on going back and forth between screens, and it get crashes randomly. will attach code link.

<img width="1067" alt="Screenshot 2023-08-01 at 16 48 58" src="https://github.com/googleads/googleads-mobile-flutter/assets/5094671/74945663-59c2-4b75-b4ef-96c7381ca1f0">

**Comment by huycozy** [comment]
Thanks for the sample repo. I checked it on both Android & iOS but it ran fine without crashing. 

<details>
<summary>Demo</summary>

https://github.com/googleads/googleads-mobile-flutter/assets/104349824/a068255e-f84c-4333-bbba-2858a1c3613d
</details>

Which iOS version and device are you seeing the crash? Can you please provide the entire output of `flutter run -v` and crash stack trace? (Please provide the crash log in text format instead of a screenshot for better readability. Thank you!)

**Comment by usman-whizpool** [comment]
see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.10.6, on macOS 12.6 21G115 darwin-x64, locale en-GB)
[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.2)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.2)
[✓] VS Code (version 1.75.1)
[✓] VS Code (version 1.67.2)
[✓] Connected device (3 available)
[✓] Network resources

• No issues found!

Also i am checking on real devices. i will share screen recording of it. iPhone 11 and iPad. 

have you tried same code i shared or you did some changes. also crash appears after some time. takes 2 - 3 minutes continues moving between screens

**Comment by huycozy** [comment]
I also tried on my physical device (iPhone 7, iOS 15.7.2) but can't reproduce it. 

>have you tried same code i shared or you did some changes

There is only one difference in adding `testDeviceIds` when initializing plugin. Even though I don't think it could be affected, you can try adding it.

```dart
  MobileAds.instance..initialize()..updateRequestConfiguration(
      RequestConfiguration(
        testDeviceIds: <String>[
          '<my device id here>'
        ],
      )
  );
```

**Comment by fahad-whizpool** [comment]
Hi @huycozy  what flutter version you are using. Because I am facing issue since Flutter v3.10.0

**Comment by usman-whizpool** [comment]
@huycozy let me try again it please

**Comment by huycozy** [comment]
>what flutter version you are using

I'm on the latest Flutter stable version: 3.10.6

**Comment by fahad-whizpool** [comment]
this issue has been resolved on latest Flutter 3.13
